common/flatpak-transaction.c

```diff
@@ -1213,6 +1215,25 @@ flatpak_transaction_set_no_deploy (FlatpakTransaction *self,
   priv->no_deploy = no_deploy;
 }
 
+/**
+ * flatpak_transaction_set_enable_diskspace_check:
+ * @self: a #FlatpakTransaction
+ * @enable_diskspace_check: whether to enable free disk space checks
+ *
+ * Sets whether the transaction should enable the checking of free disk space
+ * for the filesystem on which the current #FlatpakInstallation resides.
+ * This can be convenient for clients programs (for e.g. GNOME Software), if
+ * they want to guard against overfilling of user's disk space.
```

Comment:
Would be good to expand this a bit to say:
 * If disk space checks are enabled, the free space on the file system will be checked *before anything is downloaded*
 * The check is done when `flatpak_transaction_run()` is called, and will result in `FLATPAK_ERROR_NOT_ENOUGH_DISK_SPACE` (or whatever) being returned if it fails
 * If it fails, the transaction will be rolled back
 * Might be worth mentioning what will happen to cached downloads from the transaction; will they be deleted immediately, will they be available (still cached) if the transaction is retried, or is no particular behaviour guaranteed?
 * The check uses a conservative estimate of the final disk space consumption, and isn’t 100% accurate